lib/vector-buffers/src/variants/disk_v2/tests/basic.rs

```diff
@@ -121,7 +122,7 @@ async fn reader_exits_cleanly_when_writer_done_and_in_flight_acks() {
             }
 
             // Now acknowledge the first read, which should wake up our blocked read.
-            acker.ack(1);
+            acknowledge(first_read).await;
```

Comment:
Mostly a passing comment, but I like how this ends up making the test a lot closer to "acknowledge the entirety of the message" rather than having to manually ack a fixed number (fragile) or having to extract the event count from the record and passing that in (less fragile but still fragile).